1.  Даны две  независимые выборки. Не соблюдается условие нормальности
x1  380,420, 290
y1 140,360,200,900

    Сделайте вывод по результатам, полученным с помощью функции

In [12]:
import numpy as np
from scipy import stats

a = np.array([380,420, 290])
b = np.array([140,360,200,900])

stats.mannwhitneyu(a,b)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

In [13]:
# Значимый различий между выборками нет, т.к. pvalue > alfa

2. Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?

    1е измерение до приема препарата: 150, 160, 165, 145, 155

    2е измерение через 10 минут: 140, 155, 150,  130, 135
    
    3е измерение через 30 минут: 130, 130, 120, 130, 125

In [14]:
x_1 = np.array([150, 160, 165, 145, 155])
x_2 = np.array([140, 155, 150,  130, 135])
x_3 = np.array([130, 130, 120, 130, 125])

stats.friedmanchisquare(x_1, x_2, x_3)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

In [15]:
#Поскольку pvalue < alfa то имеются статистически значемые различия и препарат действует

3. Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.

In [16]:
stats.wilcoxon(x_1, x_2)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

In [17]:
#Статичстически значимый различий между первым и вторым измерением нет т.к. pvalue > alfa. Препарат действует по прошествии 30мин

4. Даны 3 группы  учеников плавания.
    В 1 группе время на дистанцию 50 м составляют:
    56, 60, 62, 55, 71, 67, 59, 58, 64, 67

    Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53
    
    Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54

In [18]:
a = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
b = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
c = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

stats.kruskal(a, b, c)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

In [19]:
#Поскольку pvalue > alfa статистически значимы различий между 3 группами пловцов нет. Все группы проплывают 50м примерно одинаково

5. Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%

    2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

    Поскольку дисперсия генеральной совокупности неизвестна то используем t-критерий 

In [20]:
mu = 2.5
avg = ( 2.51 + 2.35 + 2.74 + 2.56 + 2.40 + 2.36 + 2.65 + 2.7 + 2.67 + 2.34)/10
avg

2.5279999999999996

$s = \sqrt{(2.51 - 2.53)^2 + (2.35-2.53)^2 + ... + (2.34-2.53)^2}$

In [21]:
array = np.array([ 2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
def find_s (avg, array):
    sum = 0
    for i in range(len(array) - 1):
        sum += np.power((array[i] - avg), 2)
    
    return np.sqrt(sum)

s = find_s(avg, array)
s

0.4326846426671512

Используем формулу $$ t = \frac{avg-\mu}{\frac{\sigma}{\sqrt{n}}} $$

$$ t = \frac{2.53-2.5}{\frac{0.43}{\sqrt{10}}} = 0.22$$

Оприделим критиеские значение  для критерия alfa = 5% = 0.05 по таблице при степенях свободы 10-1 = 9

![Критические значения критерия Стьюдента](https://cf2.ppt-online.org/files2/slide/6/6AVb07fhBxo9I3mSvpjgwTcFJOtKHd5asWRnUr8kM/slide-3.jpg)


$t_{крит} = 2.262$

Поскольку $t = 0.22$ входит в диапазон $t_{крит}$ [-2.262; 2.262] то утверждение верное - детали изготавливаются со средним арифметическим 2,5см